devicetypes/heltun/heltun-hls01-thermostat.src/heltun-hls01-thermostat.groovy

```diff
@@ -0,0 +1,437 @@
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+	definition (name: "HELTUN HLS01 Thermostat", namespace: "HELTUN", author: "Sarkis Kabrailian", cstHandler: true, ocfDeviceType: "oic.d.thermostat") {
+		capability "Energy Meter"
+		capability "Power Meter"
+		capability "Temperature Measurement"
+		capability "Thermostat Heating Setpoint"
+		capability "Thermostat Mode"
+		capability "Thermostat Operating State"
+		capability "Voltage Measurement"    
+		capability "Configuration"
+		capability "Health Check" 
+		capability "Refresh"
+		
+		fingerprint mfr: "0344", prod: "0004", inClusters: "0x42,0x40,0x43", deviceJoinName: "HELTUN Thermostat" //model: "000A"
+	}
+	preferences {
+		input (
+			title: "HE-HLS01 | HELTUN High Load Switch",
+			description: "The user manual document with all technical information is available in support.heltun.com page. In case of technical questions please contact HELTUN Support Team at support@heltun.com",
+			type: "paragraph",
+			element: "paragraph"
+		)
+		parameterMap().each { 
+			if (it.title != null) {
+				input (
+					title: "${it.title}",
+					description: it.description,
+					type: "paragraph",
+					element: "paragraph"
+				)
+			}
+			def unit = it.unit ? it.unit : ""
+			def defV = it.default as Integer
+			def defVDescr = it.options ? it.options.get(defV) : "${defV}${unit} - Default Value" 
+			input (
+				name: it.name,
+				title: null,
+				description: "$defVDescr",
+				type: it.type,
+				options: it.options,
+				range: (it.min != null && it.max != null) ? "${it.min}..${it.max}" : null,
+				defaultValue: it.default,
+				required: false
+			)
+		}
+	}
+}
+
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	runIn(3, "checkParam")
+}
+
+def parse(String description) {
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		return zwaveEvent(cmd)
+	}
+}
+
+def checkParam() {
+	boolean needConfig = false
+	parameterMap().each {
+		if (state."$it.name" == null || state."$it.name".state == "defNotConfigured") {
+			state."$it.name" = [value: it.default as Integer, state: "defNotConfigured"]
+			needConfig = true
+		}
+		if (settings."$it.name" != null && (state."$it.name".value != settings."$it.name" as Integer || state."$it.name".state == "notConfigured")) {
+			state."$it.name".value = settings."$it.name" as Integer
+			state."$it.name".state = "notConfigured"
+			needConfig = true
+		}
+	}
+	if ( needConfig ) {
+		configParam()
+	}
+}
+
+private configParam() {
+	def cmds = []
+	for (parameter in parameterMap()) {
+	if ( state."$parameter.name"?.value != null && state."$parameter.name"?.state in ["notConfigured", "defNotConfigured"] ) { 
+			cmds << zwave.configurationV2.configurationSet(scaledConfigurationValue: state."$parameter.name".value, parameterNumber: parameter.paramNum, size: parameter.size).format()
+			cmds << zwave.configurationV2.configurationGet(parameterNumber: parameter.paramNum).format()
+			break
+		}
```

Comment:
Please fix indentations here (-1 tab)

Reply:
@KKlimczukS all the requested changes are done.
I did the same changes on HE-HT01 Handler too